Sources/PointFree/Components.swift

```diff
@@ -52,3 +52,92 @@ func gitHubLink(text: String, type: GitHubLinkType, redirect: String?) -> Node {
     ]
   )
 }
+
+func twitterShareLink(text: String, url: String, via: String? = nil) -> Node {
+
+  let tweetHref = (
+    URLComponents(string: "https://twitter.com/intent/tweet")!
+      |> (\URLComponents.queryItems) .~ [
+        URLQueryItem(name: "text", value: text),
+        URLQueryItem(name: "url", value: url),
+        via.map { URLQueryItem(name: "via", value: $0) }
+        ].compactMap(id)
+      |> ^\.url?.absoluteString
+    )
+    ?? ""
+
+  return a(
+    [
+      href(tweetHref),
+      onclick(unsafeJavascript: """
```

Comment:
why's this onclick needed?

Reply:
This allows us to open the tweet in a new window at a small size. Otherwise it just opens into a tab. 

If js is disabled it’ll still fallback to opening in a tab tho